### ISSUE TITLE
Use parser-gen `node` functionality?

### ISSUE THREAD

**Issue by hishamhm** [titan-lang/titan-v0#15]
parser-gen added support for `node` I requested here: https://github.com/vsbenas/parser-gen/issues/5 (see `pg.usenodes(value)` in [https://github.com/vsbenas/parser-gen/blob/master/README.md](README.md))

Hopefully that will slim the generated AST. :)

**Comment by hugomg** [comment]
There are two things that we are doing right now that depend on building the AST by hand and would make it hard to test this feature:

* parser-gen's generated tree stores the fields using numeric indices but we currently rely on having names for all the fields (with a custom iterator so we can make sure that we iterate them in order later on).

* Some of the tree nodes we create don't match the parser rules. For example, the `local` statement generates a declaration statement followed by an assignment statement. If we were to use the builtin ast generation we would need to write a post processing step to massage the tree in those locations (this is actually what we were doing in the very first prototype of the parser that we made, before we switched to using lpeg captures).
